cleanlab/object_detection/rank.py

```diff
@@ -0,0 +1,710 @@
+# Copyright (C) 2017-2023  Cleanlab Inc.
+# This file is part of cleanlab.
+#
+# cleanlab is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# cleanlab is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with cleanlab.  If not, see <https://www.gnu.org/licenses/>.
+import warnings
+from typing import List, Dict, Any, Optional, Tuple
+import numpy as np
+import copy
+
+# for visualizing functions
+import matplotlib.pyplot as plt
+from matplotlib.patches import Rectangle
+
+"""Methods to rank and score images in an object detection dataset (object detection data), based on how likely they
+are to contain label errors. """
+
+
+def get_label_quality_scores(
+    labels: List[Dict[Any, Any]],
+    predictions: List[np.ndarray],
+    *,
+    method: str = "subtype",
+    probability_threshold: Optional[float] = None,
+    verbose: bool = True,
+) -> np.ndarray:
+    """Returns a label quality score for each datapoint.
+
+    This is a function to compute label quality scores for object detection datasets,
+    where lower scores indicate image annotaion is less likely to be correct.
```

Comment:
```suggestion
    For object detection datasets, the label quality score for an image estimates how likely it has been correctly labeled. Lower scores indicate images whose annotation is more likely imperfect.  Annotators may have mislabeled an image because they: overlooked an object (missing annotated bounding box), chose the wrong class label for an annotated box in the correct location, or imperfectly annotated the location/edges of a bounding box. Any of these annotation errors should lead to an image with a lower label quality score.
```